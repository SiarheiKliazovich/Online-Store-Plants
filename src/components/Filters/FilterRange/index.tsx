import { FunctionComponent } from "react";
import { FilterRangeType } from "../../../types";

const FilterRange: FunctionComponent<FilterRangeType> = ({
  minLabel,
  maxLabel,
  valueFromSlider,
  minFromSlider,
  maxFromSlider,
  setFiltersFromSlider,
  setFiltersToSlider,
  valueToSlider,
  minToSlider,
  maxToSlider,
}: FilterRangeType) => (
  <div className="filter__list">
    <div className="form_control">
      {valueFromSlider !== 0 || valueToSlider !== 0 ? (
        <>
          <div className="form_control_container__time">{minLabel}</div>‚ü∑
          <div className="form_control_container__time">{maxLabel}</div>
        </>
      ) : (
        <p>Not Found</p>
      )}
    </div>
    <div className="range_container">
      <div className="sliders_control">
        <input
          id="fromSlider"
          type="range"
          value={valueFromSlider !== 0 ? valueFromSlider : minFromSlider}
          min={minFromSlider}
          max={maxFromSlider}
          onChange={(event) =>
            setFiltersFromSlider(
              valueFromSlider !== 0
                ? event.target.value
                : minFromSlider.toString()
            )
          }
        />
        <input
          id="toSlider"
          type="range"
          value={valueToSlider !== 0 ? valueToSlider : maxToSlider}
          min={minToSlider}
          max={maxToSlider}
          onChange={(event) =>
            setFiltersToSlider(
              valueToSlider !== 0 ? event.target.value : maxToSlider.toString()
            )
          }
        />
      </div>
    </div>
  </div>
);

export default FilterRange;
