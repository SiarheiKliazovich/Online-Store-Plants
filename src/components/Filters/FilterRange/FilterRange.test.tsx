import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import FilterRange from "../FilterRange";

it("renders filterRange component when valueFromSlider not equal to 0 ", () => {
  const minLabel = 1;
  const maxLabel = 2;
  const valueFromSlider = 3;
  const minFromSlider = 4;
  const maxFromSlider = 5;
  const setFiltersFromSlider = jest.fn();
  const setFiltersToSlider = jest.fn();
  const valueToSlider = 6;
  const minToSlider = 7;
  const maxToSlider = 8;
  const wrapper = render(
    <Router>
      <FilterRange
        minLabel={minLabel}
        maxLabel={maxLabel}
        valueFromSlider={valueFromSlider}
        minFromSlider={minFromSlider}
        maxFromSlider={maxFromSlider}
        setFiltersFromSlider={setFiltersFromSlider}
        setFiltersToSlider={setFiltersToSlider}
        valueToSlider={valueToSlider}
        minToSlider={minToSlider}
        maxToSlider={maxToSlider}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders filterRange component when valueFromSlider is 0 ", () => {
  const minLabel = 1;
  const maxLabel = 2;
  const valueFromSlider = 0;
  const minFromSlider = 4;
  const maxFromSlider = 5;
  const setFiltersFromSlider = jest.fn();
  const setFiltersToSlider = jest.fn();
  const valueToSlider = 6;
  const minToSlider = 7;
  const maxToSlider = 8;
  const wrapper = render(
    <Router>
      <FilterRange
        minLabel={minLabel}
        maxLabel={maxLabel}
        valueFromSlider={valueFromSlider}
        minFromSlider={minFromSlider}
        maxFromSlider={maxFromSlider}
        setFiltersFromSlider={setFiltersFromSlider}
        setFiltersToSlider={setFiltersToSlider}
        valueToSlider={valueToSlider}
        minToSlider={minToSlider}
        maxToSlider={maxToSlider}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders filterRange component valueToSlider not equal to 0 ", () => {
  const minLabel = 1;
  const maxLabel = 2;
  const valueFromSlider = 3;
  const minFromSlider = 4;
  const maxFromSlider = 5;
  const setFiltersFromSlider = jest.fn();
  const setFiltersToSlider = jest.fn();
  const valueToSlider = 6;
  const minToSlider = 7;
  const maxToSlider = 8;
  const wrapper = render(
    <Router>
      <FilterRange
        minLabel={minLabel}
        maxLabel={maxLabel}
        valueFromSlider={valueFromSlider}
        minFromSlider={minFromSlider}
        maxFromSlider={maxFromSlider}
        setFiltersFromSlider={setFiltersFromSlider}
        setFiltersToSlider={setFiltersToSlider}
        valueToSlider={valueToSlider}
        minToSlider={minToSlider}
        maxToSlider={maxToSlider}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders filterRange component valueToSlider is 0 ", () => {
  const minLabel = 1;
  const maxLabel = 2;
  const valueFromSlider = 3;
  const minFromSlider = 4;
  const maxFromSlider = 5;
  const setFiltersFromSlider = jest.fn();
  const setFiltersToSlider = jest.fn();
  const valueToSlider = 0;
  const minToSlider = 7;
  const maxToSlider = 8;
  const wrapper = render(
    <Router>
      <FilterRange
        minLabel={minLabel}
        maxLabel={maxLabel}
        valueFromSlider={valueFromSlider}
        minFromSlider={minFromSlider}
        maxFromSlider={maxFromSlider}
        setFiltersFromSlider={setFiltersFromSlider}
        setFiltersToSlider={setFiltersToSlider}
        valueToSlider={valueToSlider}
        minToSlider={minToSlider}
        maxToSlider={maxToSlider}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
