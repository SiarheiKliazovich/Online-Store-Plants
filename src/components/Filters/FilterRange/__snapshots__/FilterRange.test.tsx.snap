// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filterRange component valueToSlider is 0  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter__list"
      >
        <div
          class="form_control"
        >
          <div
            class="form_control_container__time"
          >
            1
          </div>
          ⟷
          <div
            class="form_control_container__time"
          >
            2
          </div>
        </div>
        <div
          class="range_container"
        >
          <div
            class="sliders_control"
          >
            <input
              id="fromSlider"
              max="5"
              min="4"
              type="range"
              value="3"
            />
            <input
              id="toSlider"
              max="8"
              min="7"
              type="range"
              value="8"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter__list"
    >
      <div
        class="form_control"
      >
        <div
          class="form_control_container__time"
        >
          1
        </div>
        ⟷
        <div
          class="form_control_container__time"
        >
          2
        </div>
      </div>
      <div
        class="range_container"
      >
        <div
          class="sliders_control"
        >
          <input
            id="fromSlider"
            max="5"
            min="4"
            type="range"
            value="3"
          />
          <input
            id="toSlider"
            max="8"
            min="7"
            type="range"
            value="8"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders filterRange component valueToSlider not equal to 0  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter__list"
      >
        <div
          class="form_control"
        >
          <div
            class="form_control_container__time"
          >
            1
          </div>
          ⟷
          <div
            class="form_control_container__time"
          >
            2
          </div>
        </div>
        <div
          class="range_container"
        >
          <div
            class="sliders_control"
          >
            <input
              id="fromSlider"
              max="5"
              min="4"
              type="range"
              value="3"
            />
            <input
              id="toSlider"
              max="8"
              min="7"
              type="range"
              value="6"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter__list"
    >
      <div
        class="form_control"
      >
        <div
          class="form_control_container__time"
        >
          1
        </div>
        ⟷
        <div
          class="form_control_container__time"
        >
          2
        </div>
      </div>
      <div
        class="range_container"
      >
        <div
          class="sliders_control"
        >
          <input
            id="fromSlider"
            max="5"
            min="4"
            type="range"
            value="3"
          />
          <input
            id="toSlider"
            max="8"
            min="7"
            type="range"
            value="6"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders filterRange component when valueFromSlider is 0  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter__list"
      >
        <div
          class="form_control"
        >
          <div
            class="form_control_container__time"
          >
            1
          </div>
          ⟷
          <div
            class="form_control_container__time"
          >
            2
          </div>
        </div>
        <div
          class="range_container"
        >
          <div
            class="sliders_control"
          >
            <input
              id="fromSlider"
              max="5"
              min="4"
              type="range"
              value="4"
            />
            <input
              id="toSlider"
              max="8"
              min="7"
              type="range"
              value="6"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter__list"
    >
      <div
        class="form_control"
      >
        <div
          class="form_control_container__time"
        >
          1
        </div>
        ⟷
        <div
          class="form_control_container__time"
        >
          2
        </div>
      </div>
      <div
        class="range_container"
      >
        <div
          class="sliders_control"
        >
          <input
            id="fromSlider"
            max="5"
            min="4"
            type="range"
            value="4"
          />
          <input
            id="toSlider"
            max="8"
            min="7"
            type="range"
            value="6"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders filterRange component when valueFromSlider not equal to 0  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter__list"
      >
        <div
          class="form_control"
        >
          <div
            class="form_control_container__time"
          >
            1
          </div>
          ⟷
          <div
            class="form_control_container__time"
          >
            2
          </div>
        </div>
        <div
          class="range_container"
        >
          <div
            class="sliders_control"
          >
            <input
              id="fromSlider"
              max="5"
              min="4"
              type="range"
              value="3"
            />
            <input
              id="toSlider"
              max="8"
              min="7"
              type="range"
              value="6"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="filter__list"
    >
      <div
        class="form_control"
      >
        <div
          class="form_control_container__time"
        >
          1
        </div>
        ⟷
        <div
          class="form_control_container__time"
        >
          2
        </div>
      </div>
      <div
        class="range_container"
      >
        <div
          class="sliders_control"
        >
          <input
            id="fromSlider"
            max="5"
            min="4"
            type="range"
            value="3"
          />
          <input
            id="toSlider"
            max="8"
            min="7"
            type="range"
            value="6"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
